File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/intro/ui/SplashActivity.java
package com.marcosvbras.robomarket.flows.intro.ui;

import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;

import com.marcosvbras.robomarket.R;
import com.marcosvbras.robomarket.app.Activity;
import com.marcosvbras.robomarket.databinding.ActivitySplashBinding;
import com.marcosvbras.robomarket.flows.intro.viewmodel.SplashViewModel;
import com.marcosvbras.robomarket.flows.intro.viewmodel.SplashViewModelFactory;

public class SplashActivity extends Activity {

    private ActivitySplashBinding activityBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        activityBinding = DataBindingUtil.setContentView(this, R.layout.activity_splash);
        activityBinding.setViewModel(createViewModel());
        activityBinding.executePendingBindings();
    }

    private SplashViewModel createViewModel() {
        return ViewModelProviders.of(this, new SplashViewModelFactory(this)).get(SplashViewModel.class);
    }

    @Override
    protected void onResume() {
        super.onResume();
        activityBinding.getViewModel().authenticate();
    }
}

File: project/app/src/main/java/com/marcosvbras/robomarket/flows/intro/ui/SplashActivity.kt
package com.marcosvbras.robomarket.flows.intro.ui

import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import com.marcosvbras.robomarket.R
import com.marcosvbras.robomarket.app.BaseActivity
import com.marcosvbras.robomarket.databinding.ActivitySplashBinding
import com.marcosvbras.robomarket.flows.intro.viewmodel.SplashViewModel
import com.marcosvbras.robomarket.flows.intro.viewmodel.SplashViewModelFactory

class SplashActivity : BaseActivity() {

    private var activityBinding: ActivitySplashBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN)
        activityBinding = DataBindingUtil.setContentView(this, R.layout.activity_splash)
        activityBinding?.viewModel = createViewModel()
        activityBinding?.executePendingBindings()
    }

    private fun createViewModel(): SplashViewModel {
        return ViewModelProviders.of(this, SplashViewModelFactory(this)).get(SplashViewModel::class.java)
    }

    override fun onResume() {
        super.onResume()
        activityBinding?.viewModel?.authenticate()
    }
}